### DISCUSSION TITLE
How to retrieve the energy (E) for public cloud services that surface no power, nor CPU metrics?

### DISCUSSION THREAD

**Discussion by vmeoc** [Green-Software-Foundation/sci#347] [Q&A]
Hi all,
I have seen across different use cases how to determine the Energy E for applications hosted in public cloud services where the CPU metric is available since it get turned in kwh with the tdp model. But many public cloud services does not provide such metrics. How to measure the energy when there's no power, nor CPU metric to rely on?
For example, the [AWS API Gw is very scarce ](https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-metrics-and-dimensions.html)in term of metrics. How to calculate E for AWS API Gw?
Same question for AWS Lambda or [AWS SQS ](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-available-cloudwatch-metrics.html).

**Comment by Henry-WattTime** [comment]
Hello @vmeoc 

We've started to collect recourses on calculating E here:
https://sci-data.greensoftware.foundation/E/
Do you have CPU utilization? Then  you might be able to use some of the approaches here to estimate.

Generally, the cloud providers are becoming increasing transparent around this issue.

If you have any other resources, we'd be excited to include them. We're in the early stages of developing the guidance documentation. 

Thanks, 
Working Group

**Reply by MarcePSS** [reply]
Hi @Henry-WattTime , what is the updated link? the link tells me that it cannot connect to the sci-data server.

**Reply by Henry-WattTime** [reply]
Apologies for the deadlink! Try this: https://sci-guide.greensoftware.foundation/E/
The case studies might also be useful to see how they calculated this data: https://sci-guide.greensoftware.foundation/CaseStudies

**Comment by ArneTR** [comment]
Hey @vmeoc,

for AWS Lambda and some other of their opaque products you can find the methodology approach of the CloudCarbonFootprint tool here: https://www.cloudcarbonfootprint.org/docs/methodology/#a-note-on-aws-lambda-compute-estimates

They have coefficients, same as @Henry-WattTime GSF link, per vCPU. Lambda is then estimated through the linkage of memory and CPU.
Find the coeffcients they actually use in the source here: https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/blob/trunk/packages/aws/src/lib/Lambda.ts

@Henry-WattTime The coefficients on the GSF page look a bit dated. 2.1 is according to the CloudCarbonFootprint team only for SandBridge CPUs (compare https://github.com/cloud-carbon-footprint/cloud-carbon-footprint/blob/trunk/packages/aws/src/domain/AwsFootprintEstimationConstants.ts#L44).

Maybe interesting to update the source on the GSF sci-guide? Cloud Jewels are quite old and to my knowledge never received any update any more ...?

**Reply by mrchrisadams** [reply]
There's a new paper from Facebook that might shed some more light on the assumptions for serverless. I've linked to in the discussion topic below:

https://github.com/orgs/Green-Software-Foundation/discussions/78
